### PR TITLE
AmazonS3: allow not implemented versioning

### PR DESCRIPTION
**Pull request by Erikvv**
In case the S3 implementation does not implement versioning, set it to
false. Versioning was introduced in Nexcloud in commit
https://github.com/nextcloud/server/commit/09ffac5e6dd5355c9aaf49c098942fa1e4fbed25

This is needed for compatibility with the Storj gateway.

### CODE REVIEW

**Comment by Erikvv** [comment]
We're debating whether to change this on Storj side instead.

**Comment by Erikvv** [comment]
We would like for you to consider this change.

**Review by CarlSchwan — APPROVED** [review]
The code looks good, but I don't have a Storj gateway to test it

**Comment by Erikvv** [comment]
> The code looks good, but I don't have a Storj gateway to test it

You can sign up for free and get credentials for the Storj-hosted gateway.

Alternatively I can DM you credentials. 

We also have a few ways to set up a local independent network but that is a bit more involved. One option is https://github.com/storj/up, includes gateway-mt (multitent) but not gateway-st which you could add from https://hub.docker.com/r/storjlabs/gateway. Other option is https://github.com/storj/storj/wiki/Test-network

File: apps/files_external/lib/Lib/Storage/AmazonS3.php [hunk 1]
```diff
@@ -742,6 +741,19 @@ public function versioningEnabled(): bool {
 		return $this->versioningEnabled;
 	}
 
+	protected function versioningEnabledNotCached(): bool {
```
Comment (artonge) [inline apps/files_external/lib/Lib/Storage/AmazonS3.php:744]: Maybe more explicit:

```suggestion
	protected function getVersioningStatusFromBucket(): bool {
```

Comment (Erikvv) [reply apps/files_external/lib/Lib/Storage/AmazonS3.php:744]: Great suggestion, done.

**Review by artonge — APPROVED** [review]

**Comment by artonge** [comment]
Code looks good but did not test if. It only adds a try/catch, so it should be safe to merge.

@Erikvv I assume that you tested it on your server with you storj gateway?

**Comment by Erikvv** [comment]
> @Erikvv I assume that you tested it on your server with your storj gateway?

I have tested it and I'm now able to list and download objects.

On upload there is the error https://github.com/nextcloud/server/issues/26544 but I haven't looked into this further. I know that it worked before though.

File: apps/files_external/lib/Lib/Storage/AmazonS3.php (hunk 1 above)
Comment (artonge) [reply apps/files_external/lib/Lib/Storage/AmazonS3.php:744]: Not done apparently

Comment (Erikvv) [reply apps/files_external/lib/Lib/Storage/AmazonS3.php]: Sorry, now it is.

**Comment by Erikvv** [comment]
S3 test failed but error seems unrelated, S3 server seemed down.

**Comment by artonge** [comment]
All green, merging :)

**Comment by artonge** [comment]
Thanks for this @Erikvv ! :)

**Pull request merged**